src/main/java/rx/observables/SyncOnSubscribe.java

```diff
@@ -363,7 +373,12 @@ private boolean tryUnsubscribe() {
         }
 
         private void doUnsubscribe() {
-            parent.onUnsubscribe(state);
+            try {
+                parent.onUnsubscribe(state);
+            } catch (Throwable e) {
```

Comment:
That's sad, would be nice to have a way to test such cases.

On Tue, 26 Jan 2016, 00:44 David Karnok <notifications@github.com> wrote:

> In src/main/java/rx/observables/SyncOnSubscribe.java
> <https://github.com/ReactiveX/RxJava/pull/3644#discussion_r50760936>:
>
> > @@ -363,7 +373,12 @@ private boolean tryUnsubscribe() {
> >          }
> >
> >          private void doUnsubscribe() {
> > -            parent.onUnsubscribe(state);
> > +            try {
> > +                parent.onUnsubscribe(state);
> > +            } catch (Throwable e) {
>
> Not really, this exception has nowhere to go and hijacking the plugins is
> a nightmare.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ReactiveX/RxJava/pull/3644/files#r50760936>.
>
-- 

@artem_zin